docs/src/main/asciidoc/spring-data-rest.adoc

```diff
@@ -0,0 +1,441 @@
+
+    private String name;
+
+    private String color;
+
+
+    public Fruit() {
+    }
+
+    public Fruit(String name, String color) {
+        this.name = name;
+        this.color = color;
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+}
+----
+
+
+== Configure database access properties
+
+Add the following properties to `application.properties` to configure access to a local PostgreSQL instance.
+
+[source,properties]
+----
+quarkus.datasource.db-kind=postgresql
+quarkus.datasource.username=quarkus_test
+quarkus.datasource.password=quarkus_test
+quarkus.datasource.jdbc.url=jdbc:postgresql:quarkus_test
+quarkus.datasource.jdbc.max-size=8
+quarkus.datasource.jdbc.min-size=2
+quarkus.hibernate-orm.database.generation=drop-and-create
+----
+
+This configuration assumes that PostgreSQL will be running locally.
+
+A very easy way to accomplish that is by using the following Docker command:
+
+[source,bash]
+----
+docker run --ulimit memlock=-1:-1 -it --rm=true --memory-swappiness=0 --name quarkus_test -e POSTGRES_USER=quarkus_test -e POSTGRES_PASSWORD=quarkus_test -e POSTGRES_DB=quarkus_test -p 5432:5432 postgres:11.5
+----
+
+If you plan on using a different setup, please change your `application.properties` accordingly.
+
+== Prepare the data
+
+To make it easier to showcase some capabilities of Spring Data REST on Quarkus, some test data should be inserted into the database
+by adding the following content to a new file named `src/main/resources/import.sql`:
+
+[source,sql]
+----
+INSERT INTO fruit(id, name, color) VALUES (1, 'Cherry', 'Red');
+INSERT INTO fruit(id, name, color) VALUES (2, 'Apple', 'Red');
+INSERT INTO fruit(id, name, color) VALUES (3, 'Banana', 'Yellow');
+INSERT INTO fruit(id, name, color) VALUES (4, 'Avocado', 'Green');
+INSERT INTO fruit(id, name, color) VALUES (5, 'Strawberry', 'Red');
+----
+
+Hibernate ORM will execute these queries on application startup.
+
+== Define the repository
+
+It is now time to define the repository that will be used to access `Fruit`.
+In a typical Spring Data fashion create a repository like so:
+
+[source,java]
+----
+package org.acme.spring.data.rest;
+
+import org.springframework.data.repository.CrudRepository;
+
+public interface FruitsRepository extends CrudRepository<Fruit, Long> {
+}
+----
+
+The `FruitsRepository` above extends Spring Data's `org.springframework.data.repository.CrudRepository` which means that all of the latter's methods are
+available to `FruitsRepository`.
+
+`spring-data-jpa` will generate an implementation for this repository. Then `spring-data-rest` will generate a REST CRUD resource for it.
```

Comment:
Is there a way to opt out of REST Resource generation?

Should there be some kind of build time warning if there is a clashing user defined Resource?

Reply:
You can annotate interface or a method with either `@RepositoryRestResource(exported=false)` or `@RestResource(exported=false)`. It will still generate a JAX-RS class but it will not have any method. I don't think that should cause any clashing though.

I'm not sure if there is an easy way to produce such a warning in a current state (there might be with the new REST extension). It is a valid scenario to have two classes on a same base path. In any case that would be something to look at in the `rest-data-panache` extension. `spring-data-rest` does not generate JAX-RS endpoints.

Reply:
Good to know.

Does spring-data-rest proper have a way of disabling the exposure of the REST endpoint or is it all-or-nothing on the sense that if you have spring-data-rest, you expose all the repositories?

Reply:
By default Spring Data REST picks up almost everything everything. Then you can either disable separate repositories using the annotations that I've mentioned above or change the detection strategy in application properties (all, public, public or annotated (default), annotated).
I think we should add a similar configuration option too. But we could do it as a separate feature and add it to the `rest-data-panache` too. For now the only way to disable would be with an annotation.

Reply:
Okay. Let's address the configuration in a follow-up PR